### ISSUE TITLE
Create Kubernetes label consistency

### ISSUE THREAD

**Issue by DazWilkin** [project-akri/akri#84]
**Is your feature request related to a problem? Please describe.**

+ `akri-agent` is labeled `name` ([link](https://github.com/deislabs/akri/blob/cdd071dc577536a2c3deab3ee4b8e1a39628be25/deployment/helm/templates/agent.yaml#L13))
+ `akri-controller` is labeled `app` ([link](https://github.com/deislabs/akri/blob/cdd071dc577536a2c3deab3ee4b8e1a39628be25/deployment/helm/templates/controller.yaml#L14))

It's minor but I think consistency here would be helpful (using an arbitrary `output` spec to prove the point):

```bash
TMPL="{.items[].status.phase}"

# Agent requires `name`
LABEL="name"
kubectl get pods \
--selector=${LABEL}=akri-agent \
--output=jsonpath="${TMPL}"
Running

# Controller requires `app`
LABEL="app"
kubectl get pods \
--selector=${LABEL}=akri-controller \
--output=jsonpath="${TMPL}"
Running
```

**Is your feature request related to a way you would like Akri extended? Please describe.**

**Describe the solution you'd like**

To avoid breaking any existing dependencies on the labels, propose adding the label you'd prefer to the one that doesn't have it. Since `name` is overused (and exists as part of the metadata), propose: add `app` label to `akri-agent` and swap the `matchLabels` to use `app` too:

```YAML
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: akri-agent-daemonset
spec:
  selector:
    matchLabels:
      app: akri-agent
  template:
    metadata:
      labels:
        app: akri-agent
        name: akri-agent
```

If you're confident there aren't dependencies on the `name` label, then it could be removed.

**Describe alternatives you've considered**

N/A

**Additional context**

**Comment by bfjelds** [comment]
good idea ... i'd like agent and controller to be consistent, but also want them to be consistent with what the Kubernetes community at large expects.  is there a norm for this sort of label?  i will research, but would be interested in your take (if there is one beyond you already favoring `app`).

**Comment by bfjelds** [comment]
fwiw, i'm not aware of any **name** dependency beyond **AGENT_POD_LABEL_SELECTOR** in `test/shared_test_code.py`

**Comment by DazWilkin** [comment]
When I wrote the issue, I had a quick Google to see whether there's a protocol on label naming (and deprecation) but didn't find anything.

In my experience, `app` is quite a common convention but this would likely be valued `akri` and then a further distinguishing label to identify the component (i.e. `agent`, `controller`) perhaps called `component` or whatever you refer to these as.

**Comment by romoh** [comment]
Issue involves:
1. Decide what should be the consistent label used in Akri.
2. Make sure the selected label is used in all Akri's components and we have consistency within our codebase.

**Comment by diconico07** [comment]
Fixed by #626
